File: bk2ch07p378zoomDetailed2/ch20p680zoomCentered/MyScalableView.m


#import "MyScalableView.h"

@interface MyScalableView ()
@property (strong, nonatomic) UIImage* currentImage;
@property (strong, nonatomic) NSValue* currentSize;
@end

@implementation MyScalableView

-(id)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    
    self.backgroundColor = [UIColor blackColor];
    
    return self;
}

- (void)drawRect:(CGRect)rect
{
    NSLog(@"rect %@", NSStringFromCGRect(rect));
    NSLog(@"bounds %@", NSStringFromCGRect(self.bounds));
    NSLog(@"contents scale %f", self.layer.contentsScale);
    
    NSString* path = [[NSBundle mainBundle] pathForResource:@"earthFromSaturn" ofType:@"png"];
    UIImage* im = [[UIImage alloc] initWithContentsOfFile:path];
    [im drawInRect:rect];
    

}

@end

File: bk2ch07p378zoomDetailed2/ch20p680zoomCentered/MyScalableView.swift


import UIKit

class MyScalableView : UIView {
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.backgroundColor = .black
    }
    
    override func draw(_ rect: CGRect) {
        print("rect: \(rect); bounds: \(self.bounds); scale: \(self.layer.contentsScale)")
        
        let path = Bundle.main.path(forResource: "earthFromSaturn", ofType:"png")!
        let im = UIImage(contentsOfFile:path)!
        im.draw(in:rect)
    }
    
    
}
